API/Controllers/TournamentsController.cs

```diff
@@ -20,68 +22,89 @@ IAdminNoteService adminNoteService
     /// Submit a tournament
     /// </summary>
     /// <param name="tournamentSubmission">Tournament submission data</param>
     /// <response code="400">
-    /// If the given <see cref="tournamentSubmission"/> is malformed or
-    /// if a tournament matching the given name and ruleset already exists
+    /// The tournament submission is malformed or
+    /// a tournament matching the given name and ruleset already exists
     /// </response>
     /// <response code="201">Returns location information for the created tournament</response>
     [HttpPost]
     [Authorize(Roles = OtrClaims.Roles.User)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType<TournamentCreatedResultDTO>(StatusCodes.Status201Created)]
-    public async Task<IActionResult> CreateAsync([FromBody] TournamentSubmissionDTO tournamentSubmission)
+    public async Task<IActionResult> CreateAsync([FromBody][Required] TournamentSubmissionDTO tournamentSubmission)
     {
-        if (!ModelState.IsValid)
-        {
-            return BadRequest(ModelState.ErrorMessage());
-        }
+        // Sanitize id collections
+        tournamentSubmission.Ids = tournamentSubmission.Ids.Distinct().Where(id => id >= 1);
+        tournamentSubmission.BeatmapIds = tournamentSubmission.BeatmapIds.Distinct().Where(id => id >= 1);
 
+        #region Model Validation
+
+        // Check for unique submissions
         if (await tournamentsService.ExistsAsync(tournamentSubmission.Name, tournamentSubmission.Ruleset))
         {
-            return BadRequest(
-                $"A tournament with name {tournamentSubmission.Name} for ruleset {tournamentSubmission.Ruleset} already exists"
+            ModelState.AddModelError(
+                nameof(TournamentSubmissionDTO.Name),
+                $"A tournament with the name '{tournamentSubmission.Name}' and ruleset " +
+                $"'{tournamentSubmission.Ruleset}' already exists."
+            );
+            ModelState.AddModelError(
+                nameof(TournamentSubmissionDTO.Ruleset),
+                $"A tournament with the ruleset '{tournamentSubmission.Ruleset}' and name " +
+                $"'{tournamentSubmission.Name}' already exists."
             );
         }
 
-        if (tournamentSubmission.RejectionReason.HasValue && !User.IsAdmin())
+        // Some submission data requires admin permission
+        if (!User.IsAdmin())
         {
-            return BadRequest("Only admin users may supply a rejection reason");
+            // Check reject on submit
+            if (tournamentSubmission.RejectionReason.HasValue)
+            {
+                ModelState.AddModelError(
+                    nameof(TournamentSubmissionDTO.RejectionReason),
+                    $"The field {nameof(TournamentSubmissionDTO.RejectionReason)} may only be supplied by admin users."
+                );
+            }
+
+            // We never want unprivileged users submitting empty tournaments
```

Comment:
This implies admins can submit tournaments which have no matches. Probably shouldn't be possible regardless of permission level.

Reply:
Its not meant to be an implication, that is the intended behavior. There is a specific tournament rejection reason that exists for cases when the match data is lost / unrecoverable. In this case the tournament can be submitted this way with an empty list of mp ids so that the name and ruleset combination cannot be submitted again

Reply:
Huh okay, yeah guess that makes sense.